service/controller/resource/apiendpoint/create.go

```diff
@@ -44,24 +44,24 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 
 	{
-		apiEndpoint := apiv1alpha2.APIEndpoint{
+		apiEndpoint := apiv1alpha3.APIEndpoint{
 			Host: key.ClusterAPIEndpoint(cr),
 			Port: 443,
 		}
 
-		for _, ep := range cluster.Status.APIEndpoints {
-			if ep.Host == apiEndpoint.Host && ep.Port == apiEndpoint.Port {
-				r.logger.Debugf(ctx, "API endpoint already set")
-				r.logger.Debugf(ctx, "canceling resource")
-				return nil
-			}
+		if cluster.Spec.ControlPlaneEndpoint.Host == apiEndpoint.Host && cluster.Spec.ControlPlaneEndpoint.Port == apiEndpoint.Port {
```

Comment:
`v1alpha3` Cluster CR has changed the ControlPlaneEndpoint from `Status` to `Spec`.